### PR TITLE
Allow parallel creation of windows

### PR DESCRIPTION
**Pull request by msisov**
Nothing new here. the same patch as in crrev.com

BUG=666958

patch from issue 2755673003 at patchset 100001 (http://crrev.com/2755673003#ps100001)

### CODE REVIEW

**Comment by fred-wang** [comment]
lgtm since nothing has changed.

File: services/ui/demo/mus_demo.h [hunk 1]
```diff
@@ -46,17 +46,20 @@ class MusDemo : public service_manager::Service,
...
+  // WindowTreeData is found by looking associated WindowTreeHostMus with it and
+  // then WindowTreeData::Init is called. When the destruction of the
+  // WindowTreeHostMus is announced to MusDemo (via OnWmDisplayRemoved or
+  // OnEmbedRootDestroyed), the corresponding WindowTreeData is removed by a
+  // call to RemoveWindowTreeData.
   void AppendWindowTreeData(std::unique_ptr<WindowTreeData> window_tree_data);
-  void InitWindowTreeData(
-      std::unique_ptr<aura::WindowTreeHostMus> window_tree_host);
   void RemoveWindowTreeData(aura::WindowTreeHostMus* window_tree_host);
+  std::vector<std::unique_ptr<WindowTreeData>>::iterator FindWindowTreeData(
```
Comment (tonikitoo) [inline services/ui/demo/mus_demo.h:61]: Is not it better to return a WindowTreeData* than an iterator?

File: services/ui/demo/mus_demo_external.cc [hunk 2]
```diff
@@ -66,29 +66,20 @@ void MusDemoExternal::OnStartImpl() {
   // aura::GetDeviceScaleFactorFromDisplay().
   AddPrimaryDisplay(display::Display(0));
 
-  // TODO(tonikitoo,fwang): New windows can be launched without need to wait
-  // the respective ::OnEmbed call of the previous instance.
-  OpenNewWindow();
+  for (size_t i = 0; i < number_of_windows_; ++i)
```
Comment (tonikitoo) [inline services/ui/demo/mus_demo_external.cc:69]: maybe modernize it by using our new c++ for loops? :)

File: ui/aura/mus/window_tree_client.cc [hunk 3]
```diff
@@ -909,11 +909,9 @@ void WindowTreeClient::OnEmbed(ClientSpecificId client_id,
     window_tree_host->InitHost();
     ConfigureWindowDataFromServer(window_tree_host, *root_data);
 
-    // TODO(tonikitoo): Fix the WindowTreeClientDelegate::OnEmbed API.
-    // In external window mode, this needs not to pass the ownership of the
-    // WindowTreeHostMus instance to the delegate_. A raw pointer should
-    // surface.
-    delegate_->OnEmbed(nullptr);
+    // Pass a raw pointer to WindowTreeHostMus, which helps delegate to identify
```
Comment (tonikitoo) [inline ui/aura/mus/window_tree_client.cc:912]: Pass a raw point *of* WindowTreeHostMus?

**Review by tonikitoo — COMMENTED** [review]
lgtm w/ comments

File: services/ui/demo/mus_demo.h (hunk 1 above)
Comment (msisov) [reply services/ui/demo/mus_demo.h:61]: It won't work for void MusDemo::RemoveWindowTreeData then

**Pull request merged**